src/ir/module-utils.h

```diff
@@ -186,30 +186,34 @@ template<typename T> inline void renameFunctions(Module& wasm, T& map) {
     }
   }
   wasm.updateMaps();
-  // Update other global things.
-  auto maybeUpdate = [&](Name& name) {
-    auto iter = map.find(name);
-    if (iter != map.end()) {
-      name = iter->second;
-    }
-  };
-  maybeUpdate(wasm.start);
-  ElementUtils::iterAllElementFunctionNames(&wasm, maybeUpdate);
-  for (auto& exp : wasm.exports) {
-    if (exp->kind == ExternalKind::Function) {
-      maybeUpdate(exp->value);
-    }
-  }
-  // Update call instructions.
-  for (auto& func : wasm.functions) {
-    // TODO: parallelize
-    if (!func->imported()) {
-      FindAll<Call> calls(func->body);
-      for (auto* call : calls.list) {
-        maybeUpdate(call->target);
+
+  // Update all references to it.
+  struct Updater : public WalkerPass<PostWalker<Updater>> {
+    bool isFunctionParallel() override { return true; }
+
+    T& map;
+
+    void maybeUpdate(Name& name) {
+      auto iter = map.find(name);
+      if (iter != map.end()) {
```

Comment:
We can use C++17 fanciness to make this more pithy:
```suggestion
      if (auto iter = map.find(name); iter != map.end()) {
```
(Also we typically use `it` for iterators like this)

Reply:
About `it` vs `iter`, either sgtm, but seems like `iter` is the current majority:
```
$ grep "auto iter =" `find -name "*.cpp"` | wc -l
94
$ grep "auto it =" `find -name "*.cpp"` | wc -l
55
```

Reply:
oh interesting. I guess it's only the standard in code I frequently look at 😬